File: leetcode_0387/java/leetcode_0387.java
public class leetcode_0387 {
    class Solution {
        public int firstUniqChar(String s) {
            int[] record = new int[26];
            char[] chs = s.toCharArray();
            int index = -1;
            for(char ch:chs)
                record[ch-'a']++;
            for(int i = 0;i < chs.length;i++)
                if(record[chs[i]-'a'] == 1)
                    return i;
            return index;
        }
    }
    public static void main(String[] args) {
        String s = "loveleetcode";
//        leetcode_0387 leetcode_0387 = ;
        int res = new leetcode_0387().new Solution().firstUniqChar(s);
        System.out.println(res);
        //result
        //2
    }
}

File: leetcode_0387/cpp/leetcode_0387.cpp
//
// Created by wzy on 2022/3/7.
//

#include <iostream>
#include <vector>
#include <unordered_map>
#include <sstream>

using namespace std;

class Solution {
public:
    int firstUniqChar(string s) {
        int record[26] = {0};
        int index = -1;
        for(auto ch:s)
            record[ch - 'a']++;
        for(int i = 0;i < s.length();i++)
            if(record[s[i] - 'a'] == 1)
                return i;
        return index;
    }
};

int main(int argc,char* argv[]){
    string s = "loveleetcode";
    int res = Solution().firstUniqChar(s);
    cout<<res<<endl;
    //res
    //2
}

